crates/sui-core/src/authority_active/checkpoint_executor/mod.rs

```diff
@@ -0,0 +1,295 @@
+                        return None;
+                    }
+                },
+            }
+        }
+    }
+
+    async fn schedule_checkpoint(
+        &mut self,
+        seq_num: CheckpointSequenceNumber,
+        finished: &mut CheckpointExecutionBuffer,
+    ) {
+        let next_checkpoint = match self
+            .checkpoint_store
+            .get_checkpoint_by_sequence_number(seq_num)
+        {
+            Ok(Some(new_checkpoint)) => new_checkpoint,
+            Ok(None) => {
+                self.metrics.checkpoint_exec_errors.inc();
+                panic!(
+                    "Checkpoint {:?} does not exist in checkpoint store",
+                    seq_num,
+                );
+            }
+            Err(err) => {
+                self.metrics.checkpoint_exec_errors.inc();
+                panic!("Failed to fetch checkpoint {:?}: {:?}", seq_num, err);
+            }
+        };
+
+        // Since StateSync is guaranteed to produce checkpoint sync messages
+        // in monotonically increasing order, then a mismatch of epoch would
+        // mean that we failed to execute reconfig at the epoch boundary. This
+        // is an invalid state that should not happen under normal operation
+        assert_eq!(next_checkpoint.epoch(), self.authority_state.epoch());
+
+        let state = self.authority_state.clone();
+        let store = self.checkpoint_store.clone();
+        let metrics = self.metrics.clone();
+        self.highest_scheduled_checkpoint = Some(seq_num);
+
+        let next_committee = next_checkpoint.summary().next_epoch_committee.clone();
+        if next_committee.is_some() {
+            self.end_of_epoch = true;
+        }
+
+        finished.push_back(spawn_monitored_task!(async move {
+            while let Err(err) =
+                execute_checkpoint(next_checkpoint.clone(), state.clone(), store.clone()).await
+            {
+                error!(
+                    "Error while executing checkpoint, will retry in 1s: {:?}",
+                    err
+                );
+                tokio::time::sleep(Duration::from_secs(1)).await;
+                metrics.checkpoint_exec_errors.inc();
+            }
+
+            (seq_num, next_checkpoint, next_committee)
+        }));
+    }
+}
+
+pub async fn execute_checkpoint(
+    checkpoint: VerifiedCheckpoint,
+    authority_state: Arc<AuthorityState>,
+    checkpoint_store: Arc<CheckpointStore>,
+) -> SuiResult<Vec<TransactionEffects>> {
+    let txes = checkpoint_store
+        .get_checkpoint_contents(&checkpoint.content_digest())
+        .map_err(SuiError::from)?
+        .unwrap_or_else(|| {
+            panic!(
+                "Checkpoint contents for digest {:?} does not exist",
+                checkpoint.content_digest()
+            )
+        })
+        .into_inner();
+
+    let effects = execute_and_verify_transactions(txes, authority_state).await?;
+    Ok(effects)
+}
+
+async fn execute_and_verify_transactions(
```

Comment:
Why is this called "execute_and_verify_transactions"? What does it verify?

Reply:
Ah this needs to be renamed. Previously the function was also verifying that txn effects in cert matched what was returned by `TransactionManager`. Later found that that is handled downstream so removed the verification portion